File: backend/src/main/java/com/mini/credit/dto/caisse/CaisseResponse.java
package com.mini.credit.dto.caisse;

import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Builder(toBuilder = true)
public class CaisseResponse {
    private Long id;
    private String codeCaisse;
    private String libelle;
    private Long agenceId;
    private String agenceNom;
    private Long siteId;
    private String siteNom;
    private Long antenneId;
    private String antenneNom;
    private Long caissierResponsableId;
    private String caissierResponsableNom;
    private Long caissierAffecteId;
    private String caissierAffecteNom;
    private String devise;
    private Boolean actif;
    private BigDecimal soldeTheoriqueSessionOuverte;
    private BigDecimal dernierSoldeCloture;
    private BigDecimal soldeDisponibleActuel;
    private String statutSession;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/app/features/caisse/models/caisse-response.ts
export interface CaisseResponse {
  id: number;
  codeCaisse: string;
  libelle: string;
  agenceId?: number;
  agenceNom?: string;
  siteId?: number;
  siteNom?: string;
  antenneId?: number;
  antenneNom?: string;
  caissierResponsableId?: number;
  caissierResponsableNom?: string;
  caissierAffecteId?: number;
  caissierAffecteNom?: string;
  devise: string;
  actif: boolean;
  soldeTheoriqueSessionOuverte?: number;
  dernierSoldeCloture?: number;
  soldeDisponibleActuel?: number;
  statutSession?: string;
  createdAt: string;
  updatedAt: string;
}